### PR TITLE
chore: Refactor code quality issues

### PR DESCRIPTION
**Pull request by akshgpt7**
## Description

Hey :wave:, I'm a Developer Outreach at [DeepSource](https://deepsource.io/) and ran DeepSource analysis on my fork of the repo. It found some interesting [code quality and performance improvements](https://deepsource.io/gh/akshgpt7/immudb/) to consider.

This PR fixes a few of the issues detected for you to assess if it is right for you.
Happy to provide the tweaks separately otherwise :)

## Summary of changes

- Replace multiple `append`s with one.
- Fix unnecessary calls to Printf.
- Remove unnecessary wrapping of function call.
- Fix nil context being passed to function.
- Add `.deepsource.toml` file for continuous analysis on bug risks/performance/code-quality issues on new changes.

### CODE REVIEW

**Comment by jeroiraz** [comment]
Thanks a lot @akshgpt7, we'll review it.

**Comment by akshgpt7** [comment]
I can't seem to figure out why the test is failing. Please let me know if there's something I can fix from my side to make it pass.

File: pkg/server/options.go [hunk 1]
```diff
@@ -225,18 +223,7 @@ func (o *Options) String() string {
 	if o.Logfile != "" {
 		opts = append(opts, rightPad("Log file", o.Logfile))
 	}
```
Comment (mmeloni) [inline pkg/server/options.go:225]: IMHO, readability was better before

**Comment by akshgpt7** [comment]
Thanks @mmeloni I've reverted those changes.

**Comment by akshgpt7** [comment]
@mmeloni any updates on this?

**Comment by dmacvicar** [comment]
@akshgpt7 Thanks for the PR.
Can you exclude the addition `.deepsource.toml`?. We haven't made a decision to adopt this tool.

**Comment by akshgpt7** [comment]
@dmacvicar No worries, I've removed the file :)

**Review by jeroiraz — APPROVED** [review]

**Pull request merged**